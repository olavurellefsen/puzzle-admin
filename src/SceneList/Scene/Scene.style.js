import styled from "styled-components";

export const SceneContainer = styled.div`
  height: 300px;
  width: 400px;
  margin: 20px 20px;
  border-radius: 25px;
  box-shadow: 0 4px 8px 0 rgba(0,0,0,${props => (props.selected ? 0.4 : 0.2)});
  transition: 0.3s;
  &:hover {
    box-shadow: 0 8px 16px 0 rgba(0,0,0,0.6);
  }
  opacity: ${props => (props.selected ? 1 : 0.5)};
  background-image: url('${props => props.backgroundImage}');
  background-size: cover;
`;
SceneContainer.displayName = "SceneContainer";

export const SceneTitleForm = styled.form`
  height: 25px;
  padding: 10px;
`;
SceneTitleForm.displayName = "SceneTitleForm";

export const SceneTitle = styled.input`
  display: block;
  margin : 0 auto;
  color: white;
  text-shadow: -1px -1px 0 #000, 1px -1px 0 #000, -1px 1px 0 #000,
    1px 1px 0 #000;
  font-size: 24px;
  background-color: rgba(0, 0, 0, 0);
  border: 0px;
  text-align: center;
`;
SceneTitle.displayName = "SceneTitle";

export const SceneRightArrow = styled.div`
  font-size: 60px;
  color: gray;
  text-align: right;
  transform: translate(40px,60px);
  visibility: ${props => (props.selected ? 'visible' : 'hidden')}
`;
SceneRightArrow.displayName = "SceneRightArrow";
