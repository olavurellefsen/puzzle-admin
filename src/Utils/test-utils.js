import React from "react";
import { render } from "@testing-library/react";
import { ApolloProvider } from "react-apollo";
import MainContextStore from "../MainContextStore";
import client from "../heyBreydLocalizationClient";

const AllTheProviders = ({ children }) => {
  return (
    <ApolloProvider client={client}>
      <MainContextStore>{children}</MainContextStore>
    </ApolloProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
