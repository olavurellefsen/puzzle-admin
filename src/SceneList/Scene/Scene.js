import React, { useState, useContext, useRef } from "react";
import { capitalizeFirstLetter } from "../../Utils/Utils";
import MainContext from "../../Context";
import { SceneContainer, SceneTitleForm, SceneTitle, SceneRightArrow } from "./Scene.style";

export default function Scene(props) {
  const [state, dispatch] = useContext(MainContext);
  let { id, title, image_filename } = props.scene;
  title = capitalizeFirstLetter(title);
  const [titleField, setTitleField] = useState(title);
  const sceneTitle = useRef(null);

  //If the parent component is updated (someone else has changed the value), then update the field here
  if(title!==titleField && title!==null) {
    setTitleField(title);
  }

  const handleFormSubmit = e => {
    e.preventDefault();
    sceneTitle.current.blur();
  };
  
  const selected = state.currentScene === id;
  return (
    <SceneContainer
      selected={selected}
      onClick={() => {
        dispatch({ type: "setCurrentScene", number: id });
      }}
      backgroundImage={"images/scenes/" + image_filename}
    >
      <SceneTitleForm onSubmit={handleFormSubmit}>
        <SceneTitle
          ref={sceneTitle}
          name="title"
          type="text"
          value={titleField}
          onChange={e => setTitleField(e.target.value)}
        />
      </SceneTitleForm>
      <SceneRightArrow selected={selected}>></SceneRightArrow>
    </SceneContainer>
  );
}
