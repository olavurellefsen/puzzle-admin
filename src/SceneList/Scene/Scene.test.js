import React from "react";
import { render, cleanup } from 'test-utils';
import Scene from "./Scene";
afterEach(cleanup)

const scene = { id: 1, title: "Lorum ipsum", image_filename: "Dummy filename" };

it("should render correctly", () => {
  const { container } = render(<Scene scene={scene} />);
  expect(container).toMatchSnapshot();
});

it("should show the SceneContainer element", () => {
  const { queryByTestId } = render(<Scene scene={scene} />);
  expect(queryByTestId("SceneContainer")).toBeTruthy();
});