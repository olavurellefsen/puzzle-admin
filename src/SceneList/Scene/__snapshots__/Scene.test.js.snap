// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa cLplVt"
    data-testid="SceneContainer"
  >
    <form
      class="sc-bwzfXH icviFT"
    >
      <input
        class="sc-htpNat jYNbVO"
        name="title"
        type="text"
        value="Lorum ipsum"
      />
    </form>
    <div
      class="sc-bxivhb kMcuwI"
    >
      &gt;
    </div>
  </div>
</div>
`;
